### ISSUE TITLE
Add a sample of consuming and producing messages using function + batch

### ISSUE THREAD

**Issue by StevenPG** [spring-cloud/spring-cloud-stream-samples#239]
There aren't many/any concrete examples on the web of using the producers defined here: https://docs.spring.io/spring-cloud-stream/docs/current/reference/html/spring-cloud-stream.html#_batch_producers

My team has attempted to wire this component unsuccessfully a handful of times and the main stack-overflow page that comes up in searches has an accepted answer of basically "I just switched to the stream publisher".

**Comment by sobychacko** [comment]
@StevenPG Could you describe a bit more about your use case? We would love to fill any gaps in the framework if necessary. Once we hear your use case, we can identify whether that is an issue in the framework.

**Comment by StevenPG** [comment]
Sorry for any confusion! I couldn't find any working examples of the batch producers I linked in the first issue post.

I was able to get batch producing working in the PR linked below, but not in my own project. So it's probably an issue on our end rather than a missing/broken capability in the framework.

That said, I was able to build off the existing stream sample and opened a PR for it here https://github.com/spring-cloud/spring-cloud-stream-samples/pull/240. I think this might be useful to add, due to lack of existing samples on the web.
